types/transactions.go

```diff
@@ -186,19 +187,34 @@ func (t Transaction) marshalSiaNoSignatures(w io.Writer) {
 // ID returns the id of a transaction, which is taken by marshalling all of the
 // fields except for the signatures and taking the hash of the result.
 func (t Transaction) ID() TransactionID {
+	// Get the transaction id by hashing all data minus the signatures.
 	var txid TransactionID
 	h := crypto.NewHash()
-
-	// Encode non-signature fields into hash.
 	t.marshalSiaNoSignatures(h)
+	h.Sum(txid[:0])
 
-	// Encode signatures.
-	encoding.WriteInt(h, len((t.TransactionSignatures)))
-	for i := range t.TransactionSignatures {
-		t.TransactionSignatures[i].MarshalSia(h)
+	// Sanity check in debug builds to make sure that the ids are going to be
+	// the same.
+	if build.DEBUG {
+		verify := TransactionID(crypto.HashAll(
+			t.SiacoinInputs,
+			t.SiacoinOutputs,
+			t.FileContracts,
+			t.FileContractRevisions,
+			t.StorageProofs,
+			t.SiafundInputs,
+			t.SiafundOutputs,
+			t.MinerFees,
+			t.ArbitraryData,
+		))
+
+		if verify != txid {
+			println(txid.String())
+			println(verify.String())
+			panic("transaction id marshalling is incorrect")
```

Comment:
I don't think dumping the strings helps much here, they don't really contain any actionable information